File: fixtures/adapter-ts/call_edges/src/consumer.ts
import { UserService, formatName as fmt } from "./services";

export function run(): string {
  // Instantiation of an imported class.
  const service = new UserService();

  // Call through a renamed import — resolves to the canonical symbol.
  const name = fmt("Ada", "Lovelace");

  // Member call: recorded as a property access on the receiver's type,
  // never as a call edge to the type itself.
  service.handle();

  // Neither of these names a symbol this file can resolve, so no edge.
  setTimeout(() => {}, 0);
  console.log(name);

  return name;
}

File: fixtures/adapter-py/call_edges/app/consumer.py
from .services import UserService, format_name as fmt
from requests import get


def run():
    # Construction and invocation are spelled identically in Python; the kind
    # is decided from the resolved target, not from the name.
    service = UserService()

    # Call through a renamed import, resolving to the canonical symbol.
    name = fmt("Ada", "Lovelace")

    # Attribute call: recorded as a property access on the receiver, never as
    # a call edge to the class itself.
    service.handle()

    # Neither of these names a symbol this file can resolve, so no edge and no
    # diagnostic — there is nothing here a user could fix.
    print(name)

    # A third-party function: the call is real, but the package is not what
    # ran. The import edge already records the dependency.
    get("/health")

    return len(name)
